src/index.ts

```diff
@@ -286,7 +282,7 @@ function createRdfEmitter(program: Program) {
   function nameForModel(model: Model) {
     const intrinsic = getIntrinsicModelName(program, model);
 
-    if (!intrinsic) {
+    if (!intrinsic || intrinsic !== model.name) {

```

Comment:
With this change, nameForModel will return the model name if its different from the model name, which will generally be true if you are making a new intrinsic type.

You could declare e.g. `model int32 extends Cadl.int32` which this will not work with this test, so I will ponder a better way to do this.